src/client/mixins/linked-in.ts

```diff
@@ -0,0 +1,29 @@
+import { Page } from 'puppeteer';
+import { URL } from 'url';
+
+export class LinkedInAwareMixin {
+  public client: Page;
+
+  async validateLinkedInInsightTag(pid: number) {
+    await (this as any).waitForNetworkIdle(10000, false);
+    const requests = await (this as any).getFinishedRequests();
+    const validRequests = requests.filter(r => r.method == 'GET'
+                            && (r.url.startsWith('https://px.ads.linkedin.com/collect')
+                                || r.url.startsWith('https://dc.ads.linkedin.com/collect')));
+
+    const result = validRequests.map(req => new URL(req.url)).filter(url => url.searchParams.get('pid') !== null && url.searchParams.get('pid') == pid);
+    return result.length == validRequests.length; //// this means that all linkedin requests contained the right pid
```

Comment:
I think the length equality check here may cause false negatives in edge cases.  For instance:

A company may be in the process of transitioning between LinkedIn partner accounts (e.g. because of corporate mergers, team reorganizations, re-implementations, etc), and during the transition, may wish to fire the Insight Tag for both partner accounts on the same page.

- [ ] With that in mind, perhaps just a  `result.length >= 1`